smf/clickhouse/method_script.sh

```diff
@@ -13,10 +13,11 @@ DATASTORE="$(svcprop -c -p config/store "${SMF_FMRI}")"
 args=(
 "--log-file" "/var/tmp/clickhouse-server.log"
 "--errorlog-file" "/var/tmp/clickhouse-server.errlog"
+"--config-file" "/opt/oxide/clickhouse/config.xml"
 "--"
 "--path" "${DATASTORE}"
 "--listen_host" "$LISTEN_ADDR"
 "--http_port" "$LISTEN_PORT"
 )
 
-exec /opt/oxide/clickhouse/clickhouse server "${args[@]}" &
\ No newline at end of file
+exec /opt/oxide/clickhouse/clickhouse server "${args[@]}" &
```

Comment:
Another fun quirk I found when working with config files, is that if we don't put additional configuration files in a config.d directory, ClickHouse doesn't pull all of the default values from the default configuration file to merge with our configuration file with overrides. Additionally, you [have to run the command](https://github.com/oxidecomputer/omicron/blob/main/smf/clickhouse_server/method_script.sh#L110-L112) from within the directory containing the binary, in order for ClickHouse to merge both configuration files 😅 

That said, if we no longer want all of the defaults provided by the config file that comes bundled with ClickHouse then this comment can be ignored.

Just a heads up that with the replicated setup, we're using all of the default values from the bundled file and adding our own in an additional config file that lives in config.d. We may want to do the same here?

Reply:
There _is_ no default configuration file here. I copied the contents of the embedded config file into this one. Now that one is specified, the embedded config is ignored. That means we have the same set of specified configuration parameters as before.